pkg/events/derive/derive.go

```diff
@@ -71,6 +76,31 @@ func deriveSingleEvent(id events.ID, deriveArgs deriveArgsFunction) deriveFuncti
 	}
 }
 
+// deriveMultipleEvents create an deriveFunction to generate multiple derive trace.Events.
+// The events will be created using the original event information, the ID given and the arguments given.
+// The order of the arguments given will match the order in the event definition, so make sure the order match
+// the order of the params in the events.event struct of the event under events.Definitions.
+// If the arguments given is nil, then no event will be derived.
+func deriveMultipleEvents(id events.ID, multiDeriveArgsFunc multiDeriveArgsFunction) deriveFunction {
```

Comment:
This change and its test should go in another commit with proper git log message (add derive multiple events logic).

Reply:
Not for this PR, but, It feels to me that we could have only the deriveMultipleEvents now, right ? We can keep a deriveSingleEvent as a wrapper to a single event returned deriveMultipleEvents (just to keep compatibility).

Reply:
So you are right, you don't really need the `deriveSingleEvent` function from the abstraction point-of-view.
However, I think it is much easier to add new derive events with it if multiple derived events aren't required. We can adjust the single event to use the multi-events as a specific use-case.

Reply:
Opened a PR for this - #2383

Reply:
Merged! Thank you! You can rebase this one on top of the merged PR.